runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -778,7 +778,19 @@ ClientSessionData::getOrCreateAOTCache(JITServer::ServerStream *stream)
       if (auto aotCacheMap = TR::CompilationInfo::get()->getJITServerAOTCacheMap())
          {
          _aotCache = aotCacheMap->get(_aotCacheName, _clientUID);
```

Comment:
This `get` operation creates a new empty AOT cache with the given name (if it does not exist). Maybe we should test whether we reached the limit and refuse to create one if we did. Of course, we need a test for `_aotCache != NULL` for the next operation.